aghhh.py

```
@@ -0,0 +1,108 @@
+def factorial(n):
+    if n == 1:
+        return 1
+    else:
+        return n * factorial(n - 1)  # No base case for n <= 0, causing infinite recursion
+
+print(factorial(-5))  # Will result in a RecursionError
+
+# Function with poor error handling
+def open_file(file_path):
+    try:
+        file = open(file_path, "r")
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)